diff --git a/node_modules/react-native-vlc-media-player/VLCPlayer.js b/node_modules/react-native-vlc-media-player/VLCPlayer.js
index 9b91442..0fe9e81 100755
--- a/node_modules/react-native-vlc-media-player/VLCPlayer.js
+++ b/node_modules/react-native-vlc-media-player/VLCPlayer.js
@@ -171,7 +171,7 @@ export default class VLCPlayer extends Component {
     /* const {
      source
      } = this.props;*/
-    const source = resolveAssetSource(this.props.source) || {};
+    const source = Object.assign({}, resolveAssetSource(this.props.source) || {});

     let uri = source.uri || "";
     if (uri && uri.match(/^\//)) {
diff --git a/node_modules/react-native-vlc-media-player/android/src/main/java/com/yuanzhou/vlc/vlcplayer/ReactVlcPlayerView.java b/node_modules/react-native-vlc-media-player/android/src/main/java/com/yuanzhou/vlc/vlcplayer/ReactVlcPlayerView.java
--- a/node_modules/react-native-vlc-media-player/android/src/main/java/com/yuanzhou/vlc/vlcplayer/ReactVlcPlayerView.java
+++ b/node_modules/react-native-vlc-media-player/android/src/main/java/com/yuanzhou/vlc/vlcplayer/ReactVlcPlayerView.java
@@ -133,7 +133,7 @@ public class ReactVlcPlayerView extends SurfaceView implements

     @Override
     public void onHostPause() {
-        if (!isPaused && mMediaPlayer != null) {
+        if (!isPaused && mMediaPlayer != null && libvlc != null) {
             isPaused = true;
             isHostPaused = true;
             mMediaPlayer.pause();
@@ -495,6 +495,7 @@ public class ReactVlcPlayerView extends SurfaceView implements
         //surfaceView.removeOnLayoutChangeListener(onLayoutChangeListener);
         mMediaPlayer.release();
+        mMediaPlayer = null;
         libvlc.release();
         libvlc = null;

