diff --git a/node_modules/react-native-vlc-media-player/VLCPlayer.js b/node_modules/react-native-vlc-media-player/VLCPlayer.js
index 9b91442..0fe9e81 100755
--- a/node_modules/react-native-vlc-media-player/VLCPlayer.js
+++ b/node_modules/react-native-vlc-media-player/VLCPlayer.js
@@ -171,7 +171,7 @@ export default class VLCPlayer extends Component {
     /* const {
      source
      } = this.props;*/
-    const source = resolveAssetSource(this.props.source) || {};
+    const source = Object.assign({}, resolveAssetSource(this.props.source) || {});
 
     let uri = source.uri || "";
     if (uri && uri.match(/^\//)) {
